Mlem/App/Views/Shared/Avatar/AvatarStackView.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  AvatarStackView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 08/05/2024.
+//
+
+import SwiftUI
+
+struct AvatarStackView: View {
+    let urls: [URL?]
+    let type: AvatarType
+    
+    let spacing: CGFloat
+    let outlineWidth: CGFloat
+    
+    var showPlusIcon: Bool = false
+    
+    var body: some View {
+        HStack(spacing: 0) {
+            Spacer().aspectRatio(1 / 2, contentMode: .fit)
+            HStack(spacing: spacing) {
+                ForEach(showPlusIcon ? urls : urls.dropLast(), id: \.self) { url in
+                    avatarView(url: url)
+                        .frame(maxWidth: 0)
+                        .padding(outlineWidth)
+                        .mask {
+                            Rectangle()
+                                .subtracting(.circle.offset(x: spacing))
+                                .aspectRatio(contentMode: .fill)
+                        }
+                }
+                if showPlusIcon {
+                    plusIconView()
+                        .frame(maxWidth: 0)
+                        .padding(outlineWidth)
+                } else {
+                    avatarView(url: urls.last ?? nil)
+                        .frame(maxWidth: 0)
+                        .padding(outlineWidth)
+                }
+            }
+            Spacer().aspectRatio(1 / 2, contentMode: .fit)
+        }
+    }
+    
+    @ViewBuilder
+    func plusIconView() -> some View {
```

Comment:
*nit* per CONTRIBUTING

```suggestion
    var plusIconView: some View {
```